src/ReservoirPriceOracle.sol

```diff
@@ -353,40 +456,87 @@ contract ReservoirPriceOracle is IPriceOracle, IReservoirPriceOracle, Owned(msg.
     // should we make this recursive, meaning for a route A-B-C-D
     // besides defining A-D, A-B, B-C, and C-D, we also define
     // A-B-C, B-C-D ?
-    function setRoute(address aToken0, address aToken1, address[] calldata aRoute) external onlyOwner {
+    function setRoute(address aToken0, address aToken1, address[] memory aRoute) public onlyOwner {
         if (aToken0 == aToken1) revert RPC_SAME_TOKEN();
         if (aToken1 < aToken0) revert RPC_TOKENS_UNSORTED();
         if (aRoute.length > MAX_ROUTE_LENGTH || aRoute.length < 2) revert RPC_INVALID_ROUTE_LENGTH();
         if (aRoute[0] != aToken0 || aRoute[aRoute.length - 1] != aToken1) revert RPC_INVALID_ROUTE();
 
-        _route[aToken0][aToken1] = aRoute;
-        emit Route(aToken0, aToken1, aRoute);
-
-        // iteratively define those subroutes if undefined
-        if (aRoute.length > 2) {
-            for (uint256 i = 0; i < aRoute.length - 1; ++i) {
-                (address lLowerToken, address lHigherToken) = aRoute[i].sortTokens(aRoute[i + 1]);
+        bytes32 lSlot = _calculateSlot(aToken0, aToken1);
 
-                // if route is undefined
-                address[] memory lExisting = _route[lLowerToken][lHigherToken];
-                if (lExisting.length == 0) {
-                    address[] memory lSubroute = new address[](2);
-                    lSubroute[0] = lLowerToken;
-                    lSubroute[1] = lHigherToken;
+        // simple route
+        // gas: can store aRoute.length as a local variable
+        if (aRoute.length == 2) {
+            assembly {
+                let data := shl(248, 0x01) // 0x01 in the uppermost byte
+                sstore(lSlot, data)
+            }
+            // update price for simple routes
```

Comment:
Is this unfinished?

Reply:
I'm a lil conflicted for this one. Do you think we should call `updatePrice` when setting the route? 

On one hand, it's good that we update the price right away, so that anyone who queries the oracle right after the route setting will obtain a valid price. This can be easily mitigated by separating calling `updatePrice`. 

On the other hand, it makes the function less 'clean'. It's doing more than it's supposed to. 

I'm leaning towards NOT updating the price in when setting the route.

Reply:
Agree not updating when setting